File: src/types/productivity.ts
export interface ProductivityScoreRecord {
  snapshotDate: string;
  compositeScore: number;
  dimensionScores: ProductivityDimensionScores;
  weightBreakdown: ProductivityDimensionWeights;
  explanation?: string;
  createdAt: string;
}

export interface ProductivityDimensionScores {
  completionRate: number;
  onTimeRatio: number;
  focusConsistency: number;
  restBalance: number;
  efficiencyRating: number;
}

export interface ProductivityDimensionWeights {
  completionRate: number;
  onTimeRatio: number;
  focusConsistency: number;
  restBalance: number;
  efficiencyRating: number;
}

export interface ProductivityScoreHistoryResponse {
  scores: ProductivityScoreRecord[];
  startDate: string;
  endDate: string;
  totalScores: number;
}

export interface ProductivityScoreUpsert {
  snapshotDate: string;
  compositeScore: number;
  dimensionScores: ProductivityDimensionScores;
  weightBreakdown: ProductivityDimensionWeights;
  explanation?: string;
}

File: src-tauri/src/models/productivity.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProductivityScoreRecord {
    pub snapshot_date: String,
    pub composite_score: f64,
    #[serde(default)]
    pub dimension_scores: Value,
    #[serde(default)]
    pub weight_breakdown: Value,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub explanation: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProductivityScoreUpsert {
    pub snapshot_date: String,
    pub composite_score: f64,
    pub dimension_scores: Value,
    pub weight_breakdown: Value,
    #[serde(default)]
    pub explanation: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProductivityScoreHistoryResponse {
    pub scores: Vec<ProductivityScoreRecord>,
    pub start_date: String,
    pub end_date: String,
    pub total_scores: usize,
}
